src/nifreeze/utils.py

```diff
@@ -53,23 +54,24 @@ def linear_iterator(size: int = None, **kwargs) -> Iterator[int]:
     if size is None:
         raise TypeError("Cannot build iterator without size")
 
-    return range(size)
+    return iter(range(size))
```

Comment:
isn't `range()` a generator itself?
```suggestion
    return range(size)
```

Reply:
Sorry, I had always assumed that `range` is a generator. But it is not, its of type `range`.

I can't suggest here, adding a new comment.